### ISSUE TITLE
Question: Are there any plans to reintroduce a binary client protocol?

### ISSUE THREAD

**Issue by ousado** [rethinkdb/rethinkdb#6349]

**Comment by marshall007** [comment]
@ousado no, there are not and it is unlikely that we ever would. Can you describe the use-case?

**Comment by ousado** [comment]
@marshall007 I'm looking into using rethinkdb to store mainly binary data blobs, alongside a few secondary index fields, and hence I'd prefer to avoid unnecessary en/decoding steps, if possible.

**Comment by srh** [comment]
@ousado This has come up before, even outside the context of binary data blobs, to avoid encoding/decoding work.  It's not as much a question of there being a binary client protocol as it is a matter of parameterized queries, with parameters getting sent out-of-band.  Thus, no overhead from constructing query AST's and encoding them over and over again, and also no overhead from encoding data in-band.

If you want this I could make it happen, it would take a few weeks of work.  Send me an email if you're interested.

**Comment by grantcarthew** [comment]
Have a look at [rethinkdb-regrid](https://github.com/internalfx/rethinkdb-regrid) @marshall007 

Not as related however I also made [scalable-blob-store](https://github.com/grantcarthew/node-scalable-blob-store) which can be implemented with RethinkDB.

**Comment by oliverjanik** [comment]
C# Driver https://github.com/bchavez/RethinkDb.Driver has ReGrid which works really well. Extra speed up would be great!
